{
  "origin": "codeshovel",
  "repositoryName": "hadoop",
  "repositoryPath": "/home/shaiful/research/codeshovel/codeshovel-projects/hadoop/.git",
  "startCommitName": "HEAD",
  "sourceFileName": "DynamoDBMetadataStoreTableManager.java",
  "functionName": "extractInnerException",
  "functionId": "extractInnerException___ex-IllegalArgumentException",
  "sourceFilePath": "hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/s3guard/DynamoDBMetadataStoreTableManager.java",
  "functionStartLine": 594,
  "functionEndLine": 601,
  "numCommitsSeen": 38,
  "timeTaken": 2033,
  "changeHistory": [
    "4a700c20d553dc5336ee881719bcf189fc46bfbf",
    "ee816f1fd78b029b9efa567e8b1b62391563de14"
  ],
  "changeHistoryShort": {
    "4a700c20d553dc5336ee881719bcf189fc46bfbf": "Ymovefromfile",
    "ee816f1fd78b029b9efa567e8b1b62391563de14": "Yintroduced"
  },
  "changeHistoryDetails": {
    "4a700c20d553dc5336ee881719bcf189fc46bfbf": {
      "type": "Ymovefromfile",
      "commitMessage": "HADOOP-16520. Race condition in DDB table init and waiting threads.  (#1576). Contributed by Gabor Bota.\n\nFixes HADOOP-16349. DynamoDBMetadataStore.getVersionMarkerItem() to log at info/warn on retry\r\n\r\nChange-Id: Ia83e92b9039ccb780090c99c41b4f71ef7539d35",
      "commitDate": "11/10/19 3:08 AM",
      "commitName": "4a700c20d553dc5336ee881719bcf189fc46bfbf",
      "commitAuthor": "Gabor Bota",
      "commitDateOld": "10/10/19 10:33 PM",
      "commitNameOld": "f267917ce3cf282b32166e39af871a8d1231d090",
      "commitAuthorOld": "Xiaoyu Yao",
      "daysBetweenCommits": 0.19,
      "commitsBetweenForRepo": 1,
      "commitsBetweenForFile": 1,
      "diff": "@@ -1,8 +1,8 @@\n   public static SdkBaseException extractInnerException(\n       IllegalArgumentException ex) {\n-    if (ex.getCause() instanceof  SdkBaseException) {\n+    if (ex.getCause() instanceof SdkBaseException) {\n       return (SdkBaseException) ex.getCause();\n     } else {\n       return null;\n     }\n   }\n\\ No newline at end of file\n",
      "actualSource": "  public static SdkBaseException extractInnerException(\n      IllegalArgumentException ex) {\n    if (ex.getCause() instanceof SdkBaseException) {\n      return (SdkBaseException) ex.getCause();\n    } else {\n      return null;\n    }\n  }",
      "path": "hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/s3guard/DynamoDBMetadataStoreTableManager.java",
      "extendedDetails": {
        "oldPath": "hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/s3guard/DynamoDBMetadataStore.java",
        "newPath": "hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/s3guard/DynamoDBMetadataStoreTableManager.java",
        "oldMethodName": "extractInnerException",
        "newMethodName": "extractInnerException"
      }
    },
    "ee816f1fd78b029b9efa567e8b1b62391563de14": {
      "type": "Yintroduced",
      "commitMessage": "HADOOP-15837. DynamoDB table Update can fail S3A FS init.\nContributed by Steve Loughran.\n",
      "commitDate": "11/10/18 6:57 AM",
      "commitName": "ee816f1fd78b029b9efa567e8b1b62391563de14",
      "commitAuthor": "Steve Loughran",
      "diff": "@@ -0,0 +1,8 @@\n+  public static SdkBaseException extractInnerException(\n+      IllegalArgumentException ex) {\n+    if (ex.getCause() instanceof  SdkBaseException) {\n+      return (SdkBaseException) ex.getCause();\n+    } else {\n+      return null;\n+    }\n+  }\n\\ No newline at end of file\n",
      "actualSource": "  public static SdkBaseException extractInnerException(\n      IllegalArgumentException ex) {\n    if (ex.getCause() instanceof  SdkBaseException) {\n      return (SdkBaseException) ex.getCause();\n    } else {\n      return null;\n    }\n  }",
      "path": "hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/s3guard/DynamoDBMetadataStore.java"
    }
  }
}